homeassistant/components/withings/diagnostics.py

```diff
@@ -0,0 +1,45 @@
+"""Diagnostics support for Withings."""
+from __future__ import annotations
+
+from typing import Any
+
+from yarl import URL
+
+from homeassistant.components.webhook import async_generate_url as webhook_generate_url
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_WEBHOOK_ID
+from homeassistant.core import HomeAssistant
+
+from . import (
+    CONF_CLOUDHOOK_URL,
+    WithingsMeasurementDataUpdateCoordinator,
+    WithingsSleepDataUpdateCoordinator,
+)
+from .const import DOMAIN, MEASUREMENT_COORDINATOR, SLEEP_COORDINATOR
+
+
+async def async_get_config_entry_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+
+    webhook_url = webhook_generate_url(hass, entry.data[CONF_WEBHOOK_ID])
+    url = URL(webhook_url)
+    has_valid_external_webhook_url = url.scheme == "https" and url.port == 443
+
+    has_cloudhooks = CONF_CLOUDHOOK_URL in entry.data
+
+    measurement_coordinator: WithingsMeasurementDataUpdateCoordinator = hass.data[
+        DOMAIN
+    ][entry.entry_id][MEASUREMENT_COORDINATOR]
+    sleep_coordinator: WithingsSleepDataUpdateCoordinator = hass.data[DOMAIN][
+        entry.entry_id
+    ][SLEEP_COORDINATOR]
+
+    return {
+        "has_valid_external_webhook_url": has_valid_external_webhook_url,
+        "has_cloudhooks": has_cloudhooks,
+        "webhooks_connected": measurement_coordinator.webhooks_connected,
+        "received_measurements": list(measurement_coordinator.data.keys()),
```

Comment:
Copying a dict to a list, copies the keys by default. We can remove `.keys()`.

Reply:
does this also work with set?

Reply:
```
Python 3.11.5 (main, Aug 25 2023, 13:19:53) [GCC 9.4.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.14.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: test = {"one": 1, "two": 2}

In [2]: set(test)
Out[2]: {'one', 'two'}

In [3]: list(test)
Out[3]: ['one', 'two']
```